### ISSUE TITLE
New Pint release (0.20) fails on macos

### ISSUE THREAD

**Issue by Czaki** [hgrecco/pint#1634]
We observe that the last release of Pint causes failure of tests, but only on macOS:

https://github.com/napari/napari/actions/runs/3325250531/jobs/5497734911#step:7:859
```
  napari/_vispy/_tests/test_utils.py:38: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  napari/utils/_testsupport.py:201: in actual_factory
      viewer = ViewerClass(*model_args, **model_kwargs)
  napari/viewer.py:67: in __init__
      self._window = Window(self, show=show)
  napari/_qt/qt_main_window.py:476: in __init__
      self._qt_window = _QtMainWindow(viewer, self)
  napari/_qt/qt_main_window.py:98: in __init__
      self._qt_viewer = QtViewer(viewer, show_welcome_screen=True)
  napari/_qt/qt_viewer.py:266: in __init__
      self._add_visuals()
  napari/_qt/qt_viewer.py:445: in _add_visuals
      self.scale_bar = VispyScaleBarOverlay(
  napari/_vispy/overlays/scale_bar.py:37: in __init__
      self.reset()
  napari/_vispy/overlays/scale_bar.py:142: in reset
      self._on_unit_change()
  napari/_vispy/overlays/scale_bar.py:40: in _on_unit_change
      self._unit = get_unit_registry()(self.overlay.unit)
  napari/utils/_units.py:38: in get_unit_registry
      return pint.UnitRegistry()
  /tmp/.tox/py39-macos-pyqt5/lib/python3.9/site-packages/pint/facets/plain/registry.py:136: in __call__
      obj._after_init()
  /tmp/.tox/py39-macos-pyqt5/lib/python3.9/site-packages/pint/facets/system/registry.py:78: in _after_init
      super()._after_init()
  /tmp/.tox/py39-macos-pyqt5/lib/python3.9/site-packages/pint/facets/group/registry.py:61: in _after_init
      super()._after_init()
  /tmp/.tox/py39-macos-pyqt5/lib/python3.9/site-packages/pint/facets/plain/registry.py:297: in _after_init
      loaded_files = self.load_definitions(path, True)
  /tmp/.tox/py39-macos-pyqt5/lib/python3.9/site-packages/pint/facets/plain/registry.py:519: in load_definitions
      parsed_project = self._def_parser.parse_file(file)
  /tmp/.tox/py39-macos-pyqt5/lib/python3.9/site-packages/pint/delegates/txt_defparser/defparser.py:107: in parse_file
      return fp.parse(
  /tmp/.tox/py39-macos-pyqt5/lib/python3.9/site-packages/pint/_vendor/flexparser.py:1402: in parse
      locator(source_location, target)
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  source_location = PosixPath('/tmp/.tox/py39-macos-pyqt5/lib/python3.9/site-packages/pint/default_en.txt')
  target = 'constants_en.txt'
  
      def default_locator(source_location: StrictLocationT, target: str) -> StrictLocationT:
          """Return a new location from current_location and target."""
      
          if isinstance(source_location, pathlib.Path):
              current_location = pathlib.Path(source_location).resolve()
      
              if current_location.is_file():
                  current_path = source_location.parent
              else:
                  current_path = source_location
      
              target_path = pathlib.Path(target)
              if target_path.is_absolute():
                  raise ValueError(
                      f"Cannot refer to absolute paths in import statements ({source_location}, {target})."
                  )
      
              tmp = (current_path / target_path).resolve()
              if not is_relative_to(tmp, current_path):
  >               raise ValueError(
                      f"Cannot refer to locations above the current location ({source_location}, {target})"
                  )
  E               ValueError: Cannot refer to locations above the current location (/tmp/.tox/py39-macos-pyqt5/lib/python3.9/site-packages/pint/default_en.txt, constants_en.txt)
```

As I check, Pint does not have any CI job on macOS.

**Comment by hgrecco** [comment]
Weird, I run my test locally on macos and have not seen this problem. I will add a test for macos and come back to you.

**Comment by hgrecco** [comment]
The plot thickens. All our tests run properly on macos: https://github.com/hgrecco/pint/actions/runs/3328863411

Are you using a vanilla pint definitions file?

If yes, I wonder if this is related to how tox works or how your environment is created. I have an idea that my overcome this that I will push soon.

**Comment by Czaki** [comment]
> Are you using a vanilla pint definitions file?

I think so. I will check.

**Comment by hgrecco** [comment]
I have pushed a new version of the location resolver. Tests are currently running: https://github.com/hgrecco/pint/pull/1636

If everything works fine, would it be possible for you to test against `update_flexparser` branch?. If that works, I will merge it into master and release 0.20.1 ASAP

**Comment by Czaki** [comment]
> If everything works fine, would it be possible for you to test against `update_flexparser` branch?. If that works, I will merge it into master and release 0.20.1 ASAP

Yes. I will do PR to napari with a replaced pint in dependencies to use this branch.

**Comment by jannismain** [comment]
Had the same issue. `update_flexparser` branch fixed it 👍🏻

**Comment by ebarault** [comment]
hi, was the `update_flexparser branch` deleted ?

**Comment by Czaki** [comment]
Looks like it is removed after merge to main in  #1636

**Comment by ebarault** [comment]
We also had the same issue on Linux, it seems the fix merged into master fixed it

**Comment by hgrecco** [comment]
Just released 0.20.1 fixing this.
Thanks for reporting
